edweiss-app/components/core/Day.tsx

```diff
@@ -49,6 +49,45 @@ const RenderList = ({ title, items, color, itemKey }: { title: string; items: an
     );
 
 // Main Day component
+
+// Fonction de rendu des périodes, prenant un tableau avec les détails calculés
+const renderPeriods = ({
```

Comment:
You are using a function that returns components; why not write the function as a component itself?